### ISSUE TITLE
Internationalify

### ISSUE THREAD

**Issue by moritzschaefer** [sjtug/SJTUThesis#139]
Would be nice if this would be all international(english) (including Issues)

**Comment by scozv** [comment]
Hi, @moritzschaefer 

Are you preparing the graduation thesis for [SJTU](http://en.sjtu.edu.cn/) ?
This `repo` is a `LaTeX` template and  is mainly used for SJTU degree thesis.

And, for me, I will only use this as a template, I am not able to configure the detail implementation of the 
style in this template, I just want to focus the **content**.

There are lots of `LaTeX` templates that you can find from the Google, and different templates have different style.  [such as here](http://scotv.github.io/help/2015/08/12/introduction-of-building-documents-with-latex#section-2).

To summarize, 
- IF you want to apply this `SJTUThesis` template for your thesis, I can help to write an English `READMD.md` to list the steps (1, 2, 3 ...) showing how can we build our thesis from this template
- ELSE IF you want to know the `LaTeX` style implementation of a template, I think you may find help from a English `LaTeX` template. Because translating this `repo` to English version entirely  is time-consuming. Anyway I can involve your discussion if you need help.
- ELSE `throw NotImplementionError`

**Comment by moritzschaefer** [comment]
Hey, thanks for the support. Translating the README would be a good help already!
I right now got stuck with a problem where all cites are asigned 0 for example.

**Comment by scozv** [comment]
# What is SJTUThesis?

SJTUThesis is an _unofficial_ XeLaTeX template for preparing bachelor, master, or doctor thesis in Shanghai Jiao Tong University.

The generated thesis example PDF file can be found at `README.pdf` (this file has been removed, see #26 contributing PDF into repo will increase the repo size) . The user guide will be found at  [SJTUThesis Wiki](https://github.com/weijianwen/SJTUThesis/wiki)，welcome for contribution。

## User Guide

`SJTUThesis` can be compile at your local machine, or on the Cloud service.

### System Requirement of Locally Compile

#### TeX Distribution

SJTUThesis requires XeTeX。[TeXLive](https://www.tug.org/texlive/)  distributed on 2014 and 2015,  [MacTeX](https://www.tug.org/mactex/)  released distribution can be used for compile this Template.  CTeX(2.9.2)  in Windows can be also compile  this successfully.  

[Babun](http://babun.github.io/) is recommended for **Windows User** as the `bash` terminal, which contains `git`, `GNUmake` and `perl` by default. 

#### Fonts

Chinese contents depend on four Adobe Simplified Chinese fonts, and English depends on TeX Gyre Termes font. Tex Gyre Termes can be found from [CTAN](http://www.ctan.org/tex-archive/fonts/tex-gyre/fonts/opentype/public/tex-gyre) . However, in order NOT to violate the font license, we don't provide AdobeSongStd, AdobeKaitiStd, AdobeHeitiStd or AdobeFangsongStd. (see #74  )

### Cloud Compiling by ShareLaTeX

[ShareLaTeX](https://www.sharelatex.com/) can be used for compiling the SJTUThesis 0.8.1. 
Copy the page named [SJTUThesis-0.8](https://www.sharelatex.com/project/56701ac8d63cd1d9646a622f) and then start to write your thesis.

For details, please read [Usage](https://github.com/weijianwen/SJTUThesis/wiki/Usage).

### Get the Template

Determine the `git branch` you will use depending on your system environment. `clone` this repo or download the stable distribution package.

#### Git Clone

```
# clone a new copy
cd
git clone https://github.com/weijianwen/SJTUThesis.git
# OR update the latest code from Github
git pull origin master
# OR update to your fork repo
git pull upstream master
```

#### Release Package

SJTUThesis provides multiple stale release package, the are used under different TeXLive distribution and different  `bib` compiler: 

| TeXLive distribution | bibliography  compiler | SJTUThesis version |
| --- | --- | --- |
| 2015 | biber+biblatex+caspervector | [0.9.3](https://github.com/weijianwen/SJTUThesis/releases/tag/0.9.3) |
| 2015 | bibtex+GBT7714-2005NLang.bst | [0.7.7](https://github.com/weijianwen/SJTUThesis/releases/tag/0.7.7) |
| 2014 | biber+biblatex+caspervector | [0.8.7](https://github.com/weijianwen/SJTUThesis/releases/tag/0.8.7) |
| 2013 | bibtex+GBT7714-2005NLang.bst | [0.7.1](https://github.com/weijianwen/SJTUThesis/releases/tag/0.7.1) |

### Compiling the SJTUThesis

```
# compile the SJTUThesis on --watch mode
make pvc
# compile, build and release the final PDF
make clean thesis.pdf
```

若需要生成用于提交盲审的论文(隐去作者、导师等信息)，可在`thesis.tex`中为`sjtuthesis`文档类添加`review`选项。 若需要生成包含“原创性声明扫描件”和“授权书”签名扫描件的学位论文，请将扫描件分别保存为`pdf/origignal.pdf`和`pdf/authorization.pdf`，然后添加`submit`选项重新编译模板。

#### Windows User Guide

Double click the `complie.bat`, the final PDF file will be named as `thesis.pdf`.

### Word Count

```
make wordcount
```

### Diagnosis

```
# compile manully when fail to `make`
xelatex -no-pdf thesis
biber --debug thesis
xelatex thesis
xelatex thesis
```

## Feedback

We are hearing from:
- [issue page of Github](https://github.com/weijianwen/SJTUThesis/issues)
- [SJTU BBS](https://bbs.sjtu.edu.cn/bbsdoc?board=TeX_LaTeX)

## Further Plan
- 精简代码；
- 改进开章页设计；

## License

上海交通大学校徽图片(`sjtulog.png`)和横幅图片(`sjtubanner.png`)的版权归原作者所有。其他部分使用 [Apache License 2.0](LICENSE) 授权。

**Comment by scozv** [comment]
What is your LaTeX environment, have u tried 

```
make clean thesis.pdf
```

see English README above

**Comment by scozv** [comment]
OR, have a try:

```
xelatex -no-pdf thesis
biber --debug thesis
xelatex thesis
xelatex thesis
```

**Comment by moritzschaefer** [comment]
Thanks for your help! Though it didn't help me solve my problem unfortunately: The indices of the references are still all zero. 

On top some of the references to sections in my own paper appear blank and I can't find any logfile, that tells me which references didn't work, do you have an idea where to look?

**Comment by scozv** [comment]
I once had the problem the indices of `bib` showing as `[?]`, to fix it, I have to run `xelatex` multiple times. You may check the `*.aux` file, which will tell your whether the `bib` is compiled or not:

```
# from diss.aux
\bibcite{IEEE-1363}{{6}{2000}{{IEEE Std 1363-2000}}{{}}}
\bibcite{chen2007ewi}{{7}{2007}{{Chen et al.}}{{}}}
\bibcite{kocher99}{{8}{1999}{{Kocher et al.}}{{}}}
```

Could you please tell me your compile command, and is there any output log in your terminal?

In my previous blog, I listed this issue that indices showing as `[?]`:

http://scotv.github.io/help/2015/08/12/introduction-of-building-documents-with-latex#section-6

And, I also provided a template

http://scotv.github.io/help/2015/08/12/introduction-of-building-documents-with-latex#section-9

**Comment by moritzschaefer** [comment]
It was because I used package tabularx... Weird but at least I know what to
do.
I'm all set up then.
Do you save the English readme as maybe Readme.english for future
internationals?

Scott notifications@github.com schrieb am Sa., 7. Mai 2016, 01:57:

**Comment by scozv** [comment]
hi @moritzschaefer, please wait for the PR acceptance

**Comment by weijianwen** [comment]
The PR has been accepted in 9e41de88c13 . You can check it on the dev branch, which will be available on the master in the next release.

Feedbacks on using SJTUTheis in non-Chinese thesis writing is welcome.

**Comment by scozv** [comment]
@weijianwen  

by the way, do you know the [`gitbook`](https://github.com/GitbookIO/gitbook), a documentation system supporting i18n.

see the example: http://scotv.github.io/algo-wiki/ 

**However**, I am **NOT** recommending `gitbook` for this repo, 
**because it is weird that we use a docs system `gitbook` to explain another docs system `LaTeX`.**

I also noticed you and other people were discussing #98 that using Github wiki to replace the `README`.

In my opinion,  using `README.md`, `README_en-US.md`, etc is enough for a `LaTeX` template repo, because we only need to list the steps of building the PDF file.

And, also, for `github-page`, there is another solution for i18n, named [polyglot](http://untra.github.io/polyglot/), but, I still **NOT** recommend it.
